src/Shipping/ShippingZone.php

```diff
@@ -78,6 +78,79 @@ public function get_shipping_methods_for_country( string $country_code ): array
 		return ! empty( $this->methods_countries[ $country_code ] ) ? array_values( $this->methods_countries[ $country_code ] ) : [];
 	}
 
+	/**
+	 * Gets the shipping rates for the given country.
+	 *
+	 * @param string $country_code
+	 *
+	 * @return array[] A multidimensional array of shipping rates for the given country. {
+	 *     Array of shipping method arguments.
+	 *
+	 *     @type string $country  The shipping method country.
+	 *     @type string $method   The shipping method ID.
+	 *     @type string $currency The currency which the shipping rate is in. Defaults to the store currency.
+	 *     @type float  $rate     The cost of the shipping method. Only if the method is flat-rate or local pickup.
+	 *     @type array  $options  Array of options for the shipping method (varies based on the method type). {
+	 *         Array of options for the shipping method.
+	 *
+	 *         @type array[] $shipping_class_rates An array containing the shipping class names and their rates. Only if the method is flat-rate.
+	 *         @type float   $free_shipping_threshold The minimum order amount required to use the shipping method. Only if the method is free shipping.
+	 *
+	 *     }
+	 * }
+	 */
+	public function get_shipping_rates_for_country( string $country_code ): array {
+		$methods       = $this->get_shipping_methods_for_country( $country_code );
+		$free_shipping = self::find_free_shipping_method( $methods );
+
+		$rates = [];
+		foreach ( $methods as $method ) {
+			// We process the free shipping method separately.
+			if ( ! $method['enabled'] || self::METHOD_FREE === $method['id'] ) {
```

Comment:
❓ 

What would happen if the user only has a free shipping method for a country? For example, imagine this scenario:

Spain: Only method available is Free Shipping
Portugal: Only method available is a Flat rate

When I tried to sync the rates I get only the rate for Portugal and Spain does not have any rate.

Reply:
Oh yeah, that shouldn't be 😅  Fixed it in https://github.com/woocommerce/google-listings-and-ads/pull/1192/commits/ca42c16668e13fc286fe0a31f69bc6a4313da916